packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -189,9 +201,219 @@ const renderQueryEditorHeaderLabel = (
   return null;
 };
 
+const QueryLoader = observer(
+  (props: {
+    editorStore: QueryEditorStore;
+    applicationStore: ApplicationStore<
+      LegendQueryApplicationConfig,
+      LegendQueryApplicationPlugin
+    >;
+  }) => {
+    const { editorStore, applicationStore } = props;
+    const queryFinderRef = useRef<SelectComponent>(null);
+    const [selectedQueryID, setSelectedQueryID] = useState('');
+    const queryLoaderColumns = ['Name', 'Author', 'Version'];
```

Comment:
It's fine, but let's inline this in the component, I don't think we need to do something like this